tests/manage/pv_services/test_verify_rwo_using_dc_pod.py

```diff
@@ -0,0 +1,122 @@
+)
+from ocs_ci.ocs.resources.ocs import OCS
+from ocs_ci.ocs.resources import pod
+from ocs_ci.ocs.ocp import OCP
+from tests.helpers import wait_for_resource_state, create_pod
+
+from ocs_ci.utility.utils import TimeoutSampler
+
+log = logging.getLogger(__name__)
+
+
+@tier2
+@pytest.mark.parametrize(
+    argnames='interface',
+    argvalues=[
+        pytest.param(
+            *[constants.CEPHBLOCKPOOL],
+            marks=[
+                pytest.mark.polarion_id('OCS-896'),
+            ]
+        ),
+        pytest.param(
+            *[constants.CEPHFILESYSTEM],
+            marks=[
+                pytest.mark.polarion_id('OCS-897'),
+            ]
+        )
+    ]
+)
+class TestVerifyRwoUsingReplica2DcPod(ManageTest):
+    """
+    This test class consists of tests to verify RWO volume is exclusively
+    mounted.
+    """
+    @pytest.fixture(autouse=True)
+    def setup(
+        self, interface, pvc_factory, service_account_factory, teardown_factory
+    ):
+        """
+        Create dc pod with replica 2
+        """
+        pvc_obj = pvc_factory(interface=interface, size=3)
+        sa_obj = service_account_factory(project=pvc_obj.project)
+        pod1 = create_pod(
+            interface_type=interface, pvc_name=pvc_obj.name,
+            namespace=pvc_obj.namespace, sa_name=sa_obj.name,
+            dc_deployment=True, replica_count=2,
+            deploy_pod_status=constants.STATUS_RUNNING
+        )
+        self.name = pod1.labels['name']
+        self.namespace = pod1.namespace
+
+        dc_obj = OCP(
+            kind=constants.DEPLOYMENTCONFIG, namespace=self.namespace,
+            resource_name=self.name
+        )
+        dc_info = dc_obj.get(
+            resource_name=self.name, selector=f'app={self.name}'
+        )['items'][0]
+
+        dc_obj = OCS(**dc_info)
+        teardown_factory(dc_obj)
+
+    def wait_for_pods_and_verify(self):
+        """
+        Wait for both the pods to be created and verify only one pod is running
+        """
+        # Wait for pods
+        for pods in TimeoutSampler(
+            180, 2, func=pod.get_all_pods, namespace=self.namespace,
+            selector=[self.name], selector_label='name'
+        ):
+            if len(pods) == 2:
+                break
+
+        pods_iter = cycle(pods)
+
+        # Wait for one pod to be in Running state
+        sampler = TimeoutSampler(180, 2, next(pods_iter).get)
+        for pod_info in sampler:
+            sampler.func = next(pods_iter).get
+            if pod_info['status']['phase'] == constants.STATUS_RUNNING:
+                self.running_pod = next(pods_iter)
+                break
+
+        # Verify the other pod is not coming up Running
+        try:
+            self.pod_not_running = next(pods_iter)
+            wait_for_resource_state(
+                resource=self.pod_not_running, state=constants.STATUS_RUNNING,
+                timeout=10
+            )
+            raise UnexpectedBehaviour(
```

Comment:
it seems that this exception will be thrown in any case. Could you please clarify?

Reply:
Expecting ResourceWrongStatusException from wait_for_resource_state.
If that exception did not occur, it is considered as failure and  UnexpectedBehaviour .

In other words, in the normal flow as expected by the test case, line 101 won't be executed.

Reply:
got you :)
Please add this explanation to comment in the code

Reply:
Done